packages/mui-base/src/Progress/Root/useProgressRoot.ts

```diff
@@ -0,0 +1,81 @@
+'use client';
+import * as React from 'react';
+import { mergeReactProps } from '../../utils/mergeReactProps';
+import {
+  ProgressStatus,
+  UseProgressRootParameters,
+  UseProgressRootReturnValue,
+} from './ProgressRoot.types';
+
+function getDefaultAriaValueText(value: number | null) {
+  if (value === null) {
+    return 'indeterminate progress';
+  }
+
+  return `${value}%`;
+} /**
+ *
+ * Demos:
+ *
+ * - [Progress](https://mui.com/base-ui/react-progress/#hooks)
+ *
+ * API:
+ *
+ * - [useProgressRoot API](https://mui.com/base-ui/react-progress/hooks-api/#use-progress-root)
+ */
+function useProgressRoot(parameters: UseProgressRootParameters): UseProgressRootReturnValue {
+  const {
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby,
+    'aria-valuetext': ariaValuetext,
+    direction = 'ltr',
+    getAriaLabel,
+    getAriaValueText,
+    max = 100,
+    min = 0,
+    value,
+  } = parameters;
+
+  let state: ProgressStatus = 'indeterminate';
+  if (Number.isFinite(value)) {
+    state = value === max ? 'complete' : 'loading';
```

Comment:
I'm not sure about "loading". The progress bar doesn't have to show the loading process. Perhaps `'in-progress' | 'complete'`? (plus `| 'not-started'` for completeness)

Reply:
I don't mind renaming `'loading'` to `'in-progress'`, it's closer to the component's name
But for `'not-started'`, I think we can't really differentiate that with `'in-progress'` - `0%` could mean a process has started, but no progress has been made yet (or so little that it rounds down to 0%)

Reply:
IMO you could make the same argument for `complete` - it could be 99.9% rounded up to 100%. But I don't insist. Probably best to see with @colmtuite if this third state could be useful for styling.

Reply:
We have decided on: `'indeterminate' | 'progressing' | 'complete'`